spec/datadog/core/telemetry/collector_spec.rb

```diff
@@ -176,6 +176,48 @@
         it { is_expected.to_not include(:DD_TRACE_SAMPLE_RATE) }
       end
     end
+
+    context 'DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED' do
+      around do |example|
+        ClimateControl.modify(
+          DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED: dd_trace_remove_integration_service_names_enabled
+        ) do
+          example.run
+        end
+      end
+
+      let(:dd_trace_remove_integration_service_names_enabled) { nil }
+      context 'when set to true' do
+        let(:dd_trace_remove_integration_service_names_enabled) { 'true' }
+        it { is_expected.to include(:DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED => true) }
+      end
+
+      context 'when nil defaults to false' do
+        let(:dd_trace_remove_integration_service_names_enabled) { nil }
+        it { is_expected.to include(:DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED => false) }
+      end
+    end
+
+    context 'DD_TRACE_PEER_SERVICE_MAPPING' do
+      around do |example|
+        ClimateControl.modify(
+          DD_TRACE_PEER_SERVICE_MAPPING: dd_trace_peer_service_mapping
+        ) do
+          example.run
+        end
+      end
+
+      let(:dd_trace_peer_service_mapping) { nil }
+      context 'when set' do
+        let(:dd_trace_peer_service_mapping) { 'key:value' }
+        it { is_expected.to include(:DD_TRACE_PEER_SERVICE_MAPPING => 'key:value') }
+      end
+
+      context 'when nil is not included' do
+        let(:dd_trace_peer_service_mapping) { nil }
+        it { is_expected.to_not include(:DD_TRACE_PEER_SERVICE_MAPPING) }
+      end
```

Comment:
Minor: Should we send an empty value when not present? Omitting the key can be confusing sometimes, as you don't know if it means that the data is not being collected or it is being collected, just not being used.

Reply:
yeah that makes sense

Reply:
@marcotc See here